qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -105,7 +102,7 @@ class Grover(AmplitudeAmplifier):
     """
 
     def __init__(self,
-                 iterations: Optional[Union[int, List[int], Iterator[int], float]] = None,
+                 iterations: Optional[Union[List[int], Iterator[int], int]] = None,
```

Comment:
Would it be possible the following?
```suggestion
                 iterations: Optional[Union[Iterable[int], int]] = None,
```

Reply:
I considered it, but the logic in this class around the iterations is a bit convoluted and does check `isinstance(..., list)` and treat that specially. I didn't feel confident to make such change without understanding the logic better whether it would work with all iterables.